extensions/cms/src/cmsUtils.ts

```diff
@@ -24,9 +29,30 @@ const mssqlProvider: string = 'MSSQL';
  */
 export class CmsUtils {
 
+	private static CredentialNamespace = 'cmsCredentials';
+	private _credentialProvider: azdata.CredentialProvider;
 	private _cmsService: mssql.CmsService;
 	private _registeredCmsServers: ICmsResourceNodeInfo[];
 
+	constructor() {
+		// 1) Get a credential provider
+		// 2a) Store the credential provider for use later
+		azdata.credentials.getProvider(CmsUtils.CredentialNamespace).then(credProvider => {
+			this._credentialProvider = credProvider;
```

Comment:
So this is going to mean that we don't necessarily have a cred provider after construction. At the very least you should check that credProvider is not undefined where you use it below, but really you should be ensuring that you wait until you have a credProvider before those functions return. Two ways I can think of handling this :

1. Add an async "setup" function that fetches the credProvider
2. Save the Promise here and then await on that in the save/get Password functions below

I think I'd slightly prefer 1 as it doesn't increase the overhead of each call with the only downside is that we ensure setup is called after every instance is constructed